internal/collector/telemetry/telemetry.go

```diff
@@ -31,14 +38,70 @@ const (
 	Detailed
 )
 
+var (
+	// Command-line flags that control publication of telemetry data.
+	metricsLevelPtr  *string
+	metricsPortPtr   *uint
+	metricsPrefixPtr *string
+
+	useLegacyMetricsPtr *bool
+	useNewMetricsPtr    *bool
+	addInstanceIDPtr    *bool
+)
+
+func Flags(flags *flag.FlagSet) {
```

Comment:
Not sure why these changes are here.

Reply:
we need to move flags somewhere from `service` package in order to use `metricsLevel` from  `processor/metrics.go:serviceTagsEnabled `

Reply:
I think these levels should be part of the host if you want to expose them. Don't want to block the PR, but I think we do spread this metrics levels which is something that may cause us problems in the future.